### PR TITLE
[PT FE] Extend PyTorch Frontend with MaxPool-14 and AvgPool-14

### PR DESCRIPTION
**Pull request by p-wysocki**
### Details:
 - Extension of PTFE with AvgPool-14 and MaxPool-14
 - **Needs to wait for merge of the Core PR mentioned below**
 - Reverts https://github.com/openvinotoolkit/openvino/commit/2385c2769eee002c7e174e00a9a90e886cad1122
 - They both introduce a new ceil mode: `ov::op::RoundingType::CEIL_TORCH`
 - The new ceiling mode does not allow the last pooling in a Dimension to start in the padding area
 - Currently blocked by ticket 133929

### Related PRs
 - https://github.com/openvinotoolkit/openvino/pull/22930
 - https://github.com/openvinotoolkit/openvino/pull/22966
 - https://github.com/openvinotoolkit/openvino/pull/22796
 - https://github.com/openvinotoolkit/openvino/pull/23381

### Tickets:
 - implements CVS-122418
 - blocked by CVS-133929

### Context
https://github.com/openvinotoolkit/openvino/issues/18731

### CODE REVIEW

File: src/frontends/pytorch/src/op/avg_poolnd.cpp [hunk 1]
```diff
@@ -40,35 +40,16 @@ OutputVector translate_avg_poolnd(const NodeContext& context) {
...
-        auto pads_remaining = context.mark_node(std::make_shared<v3::Broadcast>(zero_i32, pads_diff));
-        auto padding = context.mark_node(
-            std::make_shared<v0::Concat>(OutputVector{std::move(pads_remaining), std::move(pad_values)}, 0));
-        input = context.mark_node(std::make_shared<v1::Pad>(input, padding, padding, zero, ov::op::PadMode::CONSTANT));
-        pads = Shape(pads.size(), 0);
-    }
 
     return {context.mark_node(
-        std::make_shared<v1::AvgPool>(input, strides, pads, pads, kernel, !count_include_pad, rounding_type))};
+        std::make_shared<v14::AvgPool>(input, strides, pads, pads, kernel, !count_include_pad, rounding_type))};
```
Comment (rkazants) [inline src/frontends/pytorch/src/op/avg_poolnd.cpp:52]: Do we have downgrading transformation for lowering version of `AvgPool` and `MaxPool`? That is to support plugins without v14 imlementation?

**Review by mitruska — COMMENTED** [review]
Update of the op version on the frontend side should be merged after assurance of backward compatibility via downgrade transformation and corresponding update in LPT/NNCF (if applicable).

File: tests/layer_tests/pytorch_tests/test_pooling.py [hunk 2]
```diff
@@ -19,8 +19,6 @@
              {'kernel_size': [2, 1], 'stride': [], 'padding': 0},
              ]
 
-d2_params_corner_case = [{'kernel_size': [8, 8], 'stride': [8, 4], 'padding': 1}]
```
Comment (mvafin) [inline tests/layer_tests/pytorch_tests/test_pooling.py:22]: This case should be added to `d2_params` otherwise you do not check that it works

**Review by mvafin — APPROVED** [review]

**Comment by mvafin** [comment]
Tests are failing. Is it because CPU support is not yet merged?

**Comment by p-wysocki** [comment]
> Tests are failing. Is it because CPU support is not yet merged?

That's correct. Fallback to template plugin is impossible due to evaluate method not being implemented. Implementing it would case performance degradation, so we need to wait for CPU implementation. I'm currently looking into it.

**Review by praasz — APPROVED** [review]

**Comment by mvafin** [comment]
@p-wysocki Could you also check that your changes fix #23526 ? And add the test for that case.

**Pull request closed without merging**